### PR TITLE
Initial End-to-End Testing Framework & Dependency Injection

### LINKED ISSUE #149: Create End-2-End tests

### PR DESCRIPTION
**Pull request by mohamedtahaguelzim**
### Summary:
This PR introduces essential end-to-end (E2E) testing capabilities for the activity flow and includes significant refactors to improve maintainability and testability. It prepares the project for comprehensive multi-screen testing and finally correctly implements Dependency Injection.

### Key Changes:
1. **E2E Test Setup**:
   - Added a foundational E2E test setup for activities.
   - Integrated mock data for `Activity` and `User` repositories to facilitate testing.
   - Mocked location permission checker to simulate real-world app behavior.
   - Initialized test environment using `Mockito` for repository interaction and configured dependencies with `Hilt`.

2. **Refactoring**:
   - Renamed `ActivitiesApp` to `NavGraph` to better reflect its role as a `NavHost` function.
   - Removed redundant parameters (`uid`) and comments for cleaner code.
   - Updated variable definitions to follow dependency injection best practices.

3. **Dependency Injection**:
   - Project now implements dependency injection. Different repositories implementations are used for production/testing code.

4. **Test Enhancements**:
   - Configured multi-screen testing preparations, allowing future tests to navigate seamlessly between screens.
   - Added missing test tags to important components like the sign-in screen.

### Commits Overview:
- **Commit Highlights**:
  - Initial data mock additions.
  - Refactoring of `ActivityApp` for streamlined testing.
  - Setup for comprehensive test preparation.
  - Minor test tag additions and naming convention updates.

### Next Steps:
- Replace temporary sample data with a structured mock database (`dummy_data`).
- Add new end to end tests.
- Convert and optimise some previous tests, with the now implemented Hilt Dependency Injection

### Review Notes:
Reviewers are encouraged to check the following:
- Refactoring impact on current implementation.
- Completeness of E2E test cases and view model instantiation.
- Consistency and clarity of naming conventions post-renaming.

Close #149

### CODE REVIEW

File: app/src/main/java/com/android/sample/model/map/PermissionChecker.kt [hunk 1]
```diff
@@ -0,0 +1,5 @@
+package com.android.sample.model.map
```
Comment (DAFERMOH) [inline app/src/main/java/com/android/sample/model/map/PermissionChecker.kt:1]: LGTM!

File: app/src/main/java/com/android/sample/model/profile/MockProfilesRepository.kt [hunk 2]
```diff
@@ -0,0 +1,89 @@
+package com.android.sample.model.profile
```
Comment (DAFERMOH) [inline app/src/main/java/com/android/sample/model/profile/MockProfilesRepository.kt:1]: good mocking of the profileRepository that will be used for tests and good error handling. (addLikedActivity and removeLikedActivity need to be implemented later).

File: app/src/main/java/com/android/sample/ui/authentication/SignIn.kt [hunk 3]
```diff
@@ -55,7 +55,7 @@ fun SignInScreen(navigationActions: NavigationActions, viewModel: SignInViewMode
       rememberGoogleSignInLauncher(viewModel, navigationActions, onAuthSuccess, onAuthError)
 
   Scaffold(
```
Comment (DAFERMOH) [inline app/src/main/java/com/android/sample/ui/authentication/SignIn.kt:57]: LGTM!

**Review by DAFERMOH — APPROVED** [review]
**Summary**
This PR introduces several new test modules and modifies existing functionality in the Android application.

**Review**
New Test Modules:

Added TestFirebaseModule.kt and TestRepositoryModule.kt for testing purposes using Mockito and Dagger Hilt.
Added ActivityFlowTest.kt to test end-to-end flows in the application using Compose and Hilt.
MainActivity Changes:

Refactored ActivitiesApp to NavGraph for better modularity and navigation management.
Removed unnecessary ViewModel factory comments and streamlined ViewModel initializations.
Repository and Permission Changes:

Unified the LocationPermissionChecker and PermissionChecker to enhance the permission checking mechanism.
Updated RepositoryModule to provide mock implementations for testing.
UI Changes:

Added test tags to SignInScreen components to facilitate UI testing.

**Recommendations**
Ensure that all new test cases cover the expected scenarios and edge cases.
Verify that the refactored navigation logic works seamlessly with the existing navigation flows.

File: app/src/main/java/com/android/sample/model/map/PermissionChecker.kt (hunk 1 above)
Comment (aminebenji04) [reply app/src/main/java/com/android/sample/model/map/PermissionChecker.kt:1]: I am not sure this new class u=is useful but there is no problem

**Review by aminebenji04 — APPROVED** [review]
Overal everything is good, now thanks to you we will have a really simple example on how to build tests
